### PR TITLE
refactor(chequebook): added integer validation and BigInt as type

### PR DESCRIPTION
**Pull request by vojtechsimetka**
`Number.MAX_SAFE_INTEGER` is `9007199254740991 ~= 9 * 10^16` which is just `9 BZZ`. It is reasonable to assume, that sooner or later users would want to withdraw or deposit more.

I've decided to extract this part from the #207 PR - see https://github.com/ethersphere/bee-js/pull/207#discussion_r597578613

### CODE REVIEW

**Comment by agazso** [comment]
I just realized that we only use `BigInt` as an input type here. If we used `BigInt | number` then it would work the same as before (it wouldn't be a breaking change) but it could also work well with numbers bigger than max integer, no?

**Comment by vojtechsimetka** [comment]
yeah but that could lead people to inocorrect usage. Take for an example a value just 10* bigger than `MAX_SAFE_INTEGER` and try to console log it:
```
console.log(90071992547409910)
// result   90071992547409900

console.log(90071992547409910n)
// result   90071992547409910n
```

**Comment by agazso** [comment]
> yeah but that could lead people to inocorrect usage. Take for an example a value just 10* bigger than `MAX_SAFE_INTEGER` and try to console log it:
> 
> ```
> console.log(90071992547409910)
> // result   90071992547409900
> 
> console.log(90071992547409910n)
> // result   90071992547409910n
> ```

That's a good point and we can do [input validation](https://github.com/ethersphere/bee-js/issues/202) for the number type (also check for `Number.isInteger()`), which was missing previously btw.

**Comment by vojtechsimetka** [comment]
@agazso are you still proposing to keep the `number` as input option but then have the following check?

```ts
isNumberInteger(amount: unknown) {
  return typeof amount === 'number'
      && amount > 0
      && amount < Number.MAX_SAFE_INTEGER
      && Number.isInteger(amount)
}

isBigInt(amount: unknown) {
  return typeof amount === 'bigint' && amount > 0)
}
```

**Comment by agazso** [comment]
> @agazso are you still proposing to keep the `number` as input option but then have the following check? This would then be a runtime check of corse
> 
> ```ts
> isNumberInteger(amount: unknown) {
>   return typeof amount === 'number'
>       && amount > 0
>       && amount < Number.MAX_SAFE_INTEGER
>       && Number.isInteger(amount)
> }
> 
> isBigInt(amount: unknown) {
>   return typeof amount === 'bigint' && amount > 0
> }
> ```

Yes, something along these lines (input validation is a runtime check generally). Only difference that I would do the `amount > 0`  check separately, e.g. `(isNumberInteger(amount) || isBigInt(amount)) && amount > 0)`

File: src/utils/type.ts [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+    typeof type === 'bigint' ||
+    (typeof type === 'number' &&
+      type > Number.MIN_SAFE_INTEGER &&
+      type < Number.MAX_SAFE_INTEGER &&
+      Number.isInteger(type))
+  )
+}
+
+export function assertInteger(type: unknown): asserts type is number | BigInt {
+  if (isInteger(type)) throw new TypeError('value is not integer')
```
Comment (agazso) [inline src/utils/type.ts:12]: This doesn't look correct 😄
It seems that a logical negation is missing.

File: src/utils/type.ts [hunk 2]
```diff
@@ -0,0 +1,13 @@
+export function isInteger(type: unknown): type is number | BigInt {
```
Comment (agazso) [inline src/utils/type.ts:1]: I think `value` is a more appropriate name instead of `type`

**Review by agazso — CHANGES_REQUESTED** [review]
`assertInteger` seems to do the opposite what its name suggests. Luckily the tests caught it 😄 

Other than that it looks good.

**Review by agazso — APPROVED** [review]

**Pull request merged**